src/emqtt.erl

```diff
@@ -1252,6 +1418,13 @@ delete_inflight_when_full(Packet, State) ->
         false -> {next_state, connected, State1}
     end.
 
+drop_expired(Pendings) ->
+    Now = now_(),
+    queue:filter(
```

Comment:
performance killer.
we should only need to pop out the expired ones and return as soon as the next one is not expired.
so there is no need to scan the whole queue over and over again.